src/libs/processcontext.cpp

```diff
@@ -69,7 +69,14 @@ ProcessObj *ProcessContext::createProcess(sinsp_threadinfo *ti, sinsp_evt *ev,
 
   if (parent != nullptr) {
     OID poid;
-    parent = parent->is_main_thread() ? parent : parent->get_main_thread();
+    if (!parent->is_main_thread()) {
+      sinsp_threadinfo *main = parent->get_main_thread();
+      if (main != nullptr) {
+        parent = main;
+      } else {
+        SF_DEBUG(m_logger, "Parent main thread is NULL!!");
```

Comment:
Nit: remove the exclamation marks or use a single exclamation mark only (in the log message).